fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/RelationalExpression.java

```diff
@@ -107,10 +114,14 @@ static RelationalExpression fromRecordQuery(@Nonnull PlanContext context,
         final GroupExpressionRef<? extends RelationalExpression> baseRef;
         Quantifier.ForEach quantifier;
         if (recordTypes.isEmpty()) {
-            baseRef = GroupExpressionRef.of(new FullUnorderedScanExpression(context.getMetaData().getRecordTypes().keySet()));
+            baseRef = GroupExpressionRef.of(new FullUnorderedScanExpression(context.getMetaData().getRecordTypes().keySet(), context.getMetaData().getAllIndexes().stream()
```

Comment:
As both branches have a common subexpression, maybe extract that and set a local variable beforehand. Also, you could add a new method to `MetaDataPlanContext` which gets you the index names quickly, as the index names are the key set of the bi-map `indexesByName`.